internal/pkg/cli/pipeline_init.go

```diff
@@ -359,24 +355,44 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) getGitHubOwner() error {
+	owner, err := opts.prompt.Get(
+		pipelineEnterGitHubOwnerPrompt,
+		pipelineEnterGitHubOwnerHelpPrompt,
+		nil,
 	)
+	if err != nil {
+		return fmt.Errorf("fail to get GitHub owner name: %w", err)
+	}
+	opts.GitHubOwner = owner
+	return nil
+}
 
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+func (opts *InitPipelineOpts) listGitHubRepo() ([]string, error) {
+	repos, _, err := opts.repo.List(opts.context, opts.GitHubOwner, &github.RepositoryListOptions{})
```

Comment:
What do you think about calling `git remote -v` to list the available remotes in the repo? We can parse the output and grab even the names under the owner.

For example,
```go
$ git remote -v
dmnlk	https://github.com/dmnlk/amazon-ecs-cli.git (fetch)
dmnlk	https://github.com/dmnlk/amazon-ecs-cli.git (push)
external	ssh://git.amazon.com:2222/pkg/AmazonECSCLIExternal (fetch)
external	ssh://git.amazon.com:2222/pkg/AmazonECSCLIExternal (push)
internal	ssh://git.amazon.com:2222/pkg/MadisonCliInternal (fetch)
internal	ssh://git.amazon.com:2222/pkg/MadisonCliInternal (push)
karakuse	git@github.com:efekarakus/amazon-ecs-cli.git (fetch)
karakuse	git@github.com:efekarakus/amazon-ecs-cli.git (push)
origin	https://github.com/aws/amazon-ecs-cli (fetch)
origin	https://github.com/aws/amazon-ecs-cli (push)
```

This way we can filter to only relevant repositories and get rid of the dependency to `github.com/google/go-github/github`. Otherwise, for an owner like `aws` the list can be huge.

Reply:
The result list can get filtered and matched. As talked yesterday, the only problem with this is such big owner like "aws" need more time (like 5 sec) to get all the repo name. Using `git remote -v` only get the repo info locally, which may not sync up with the upstream. I think what we want is actually the upstream repo info, from which the code pipeline directly sources.

Reply:
:+1 to removing a dependency. The pipeline workflow is going to write a local `pipeline.yml` file to the workspace which should be in the related repository. I don't think in the current state it will be possible to support `pipeline init` outside of a project/workspace. It seems like it should be safe to source repository info from the local repository.